substrate/frame/nis/src/lib.rs

```diff
@@ -515,12 +546,19 @@ pub mod pallet {
 	}
 
 	#[pallet::hooks]
-	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
-		fn on_initialize(n: BlockNumberFor<T>) -> Weight {
+	impl<T: Config> Hooks<SystemBlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(_n: SystemBlockNumberFor<T>) -> Weight {
+			let block_number = T::BlockNumberProvider::current_block_number();
 			let mut weight_counter =
 				WeightCounter { used: Weight::zero(), limit: T::MaxIntakeWeight::get() };
-			if T::IntakePeriod::get().is_zero() || (n % T::IntakePeriod::get()).is_zero() {
+			let last_processed = LastProcessedBlock::<T>::get();
+			let intake_period = T::IntakePeriod::get();
+			if intake_period.is_zero() ||
```

Comment:
```suggestion
			if 
```

saturating_sub will be always at least zero and thus, we don't need this check for zero.

Reply:
Done